knapsack/index.js

```diff
@@ -0,0 +1,372 @@
+const readline = require('readline');
+const fs = require('fs');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+    terminal: false
+});
+
+
+class Knapped {
+    constructor() {
+        this.values = [];
+        this.weights = [];
+        this.items = [];
+        this.averages = {
+            weight: 0, value: 0
+        }
+        this.capacity = 0;
+
+        this._prompt = ' \u2B55 \u0020\u0020';
```

Comment:
A comment to describe what this is may be nice, I have a guess but it looks magical without further investigation.